sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobAPITest.groovy

```diff
@@ -2232,6 +2357,38 @@ class BlobAPITest extends APISpec {
         thrown(BlobStorageException)
     }
 
+    def "Set tier tags"() {
+        setup:
+        def cc = blobServiceClient.createBlobContainer(generateContainerName())
+        def bc = cc.getBlobClient(generateBlobName()).getBlockBlobClient()
+        bc.upload(defaultInputStream.get(), defaultDataSize)
+        def t = new HashMap<String, String>()
+        t.put("foo", "bar")
+        bc.setTags(t)
+
+        when:
+        bc.setAccessTierWithResponse(new BlobSetAccessTierOptions(AccessTier.HOT).setIfTags("\"foo\" = 'bar'"), null, null)
+
+        then:
+        notThrown(BlobStorageException)
+
+        cleanup:
+        cc.delete()
+    }
+
+    def "Set tier tags fail"() {
```

Comment:
Why didn't you use the same data table as get/set tags?

Reply:
This one already had a set tier lease and set tier lease fail. I could combine those 2 into one if you prefer but it was easier for me just to add these tests like this.